src/main/java/com/nttdata/estore/controllers/AdminController.java

```diff
@@ -1,30 +1,54 @@
 package com.nttdata.estore.controllers;
+
 import com.nttdata.estore.entities.Admin;
-import com.nttdata.estore.repositories.AdminRepository;
 import com.nttdata.estore.services.AdminService;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.beans.factory.annotation.Qualifier;
-import org.springframework.stereotype.Controller;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.ResponseBody;
-
-@Controller    // This means that this class is a Controller
-@RequestMapping(path="/admin")
-public class AdminController {
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
 
-    //@Autowired
-    //private AdminRepository adminRepository;
+@RestController
+@CrossOrigin
+public class AdminController {
 
     @Autowired
     private AdminService adminService;
 
-    @GetMapping(path="/allAdmins")
-    public @ResponseBody Iterable<Admin> getAllAdmins() {
-        // This returns a JSON or XML with the users
+    @GetMapping(path = "/admins")
+    public @ResponseBody
+    Iterable<Admin> getAllAdmins() {
         return adminService.findAll();
     }
 
+    @GetMapping(path = "/admins/{id}")
+    public ResponseEntity getAdmin(@PathVariable("id") int id) {
+        Admin admin = adminService.getAdmin(id);
+        if (admin == null) {
+            return new ResponseEntity("No Admin found for ID " + id, HttpStatus.NOT_FOUND);
+        }
+        return new ResponseEntity(admin, HttpStatus.OK);
+    }
+
+    @DeleteMapping(path = "/admins/{id}")
+    public ResponseEntity deleteAdmin(@PathVariable int id) {
+        adminService.deleteAdmin(id);
+        return new ResponseEntity(id, HttpStatus.OK);
+    }
 
+    @PostMapping(path = "/admins")
+    public ResponseEntity createAdmin(@RequestBody Admin admin) {
+        adminService.createAdmin(admin);
+        return new ResponseEntity(admin, HttpStatus.OK);
+    }
 
-}
+    @PutMapping(path = "/admins/{id}")
+    public ResponseEntity updateAdmin(@PathVariable int id, @RequestBody Admin admin) {
+        Admin oldAdmin = adminService.getAdmin(id);
+        oldAdmin = admin;
+        adminService.createAdmin(oldAdmin);
```

Comment:
Nu inteleg logica acestei metode :( Prima data citesti admin-ul din baza de date(in functie de id) si il pui in oldAdmin.
Apoi suprascrii oldAdmin cu admin-ul primit de pe request(deci se pierde valoarea citita in linia 46) iar intr-un final creezi un admin cu ajutorul oldAdmin(deci admin-ul primit de pe request).